src/carriers/zfp_portmonitor/zfpPortmonitor.cpp

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (C) 2016 iCub Facility
+ * Authors: Nicolo' Genesio
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <math.h>
+#include <algorithm>
+#include <yarp/sig/Image.h>
+#include <yarp/os/LogStream.h>
+#include "zfp.h"
+
+#include "zfpPortmonitor.h"
+
+using namespace yarp::os;
+using namespace yarp::sig;
+
+
+bool ZfpMonitorObject::create(const yarp::os::Property& options)
+{
+   shouldCompress = (options.find("sender_side").asBool());
+   compressed=NULL;
+   decompressed=NULL;
+   //yDebug()<< ((shouldCompress) ? "compressing..." : "decompressing...");
+   return true;
+}
+
+void ZfpMonitorObject::destroy(void)
+{
+    //yDebug()<<"destroyed";
+}
+
+bool ZfpMonitorObject::setparam(const yarp::os::Property& params)
+{
+    return false;
+}
+
+bool ZfpMonitorObject::getparam(yarp::os::Property& params)
+{
+    return false;
+}
+
+bool ZfpMonitorObject::accept(yarp::os::Things& thing)
+{
+    if(shouldCompress){
+        ImageOf<PixelFloat>* img = thing.cast_as< ImageOf<PixelFloat> >();
+        if(img == NULL) {
+            yError()<<"ZfpMonitorObject: expected type ImageOf<PixelFloat> in sender side, but got wrong data type!";
+            return false;
+        }
+    }
+    else{
+        Bottle* bt= thing.cast_as<Bottle>();
+        if(bt==NULL){
+            yError()<<"ZfpMonitorObject: expected type Bottle in receiver side, but got wrong data type!";
+            return false;
+        }
+
+    }
+
+
+    return true;
+}
+
+yarp::os::Things& ZfpMonitorObject::update(yarp::os::Things& thing)
+{
+
+   if(shouldCompress) {
+       if(compressed){
+           free(compressed);
+           compressed=NULL;
+       }
+        ImageOf<PixelFloat>* img = thing.cast_as< ImageOf<PixelFloat> >();
+        // .... buffer, len
+        int sizeCompressed;
+        compress((float*)img->getRawImage(), compressed, sizeCompressed, img->width(),img->height(),1e-3);
+        if(!compressed){
+            yError()<<"Failed to compress, exiting...";
```

Comment:
Prepending the name of the plugin/device to this error message (as in `"ZfpMonitorObject: Failed to compress, exiting."`) will simplify the job of any user that is tryng to process this error message.